lib/matplotlib/_mathtext.py

```diff
@@ -2858,6 +2876,42 @@ def binom(self, s, loc, toks):
         return self._genfrac('(', ')', 0.0, self._MathStyle.TEXTSTYLE,
                              num, den)
 
+    def _genset(self, state, value, symbol, shift):
+        thickness = state.font_output.get_underline_thickness(
+            state.font, state.fontsize, state.dpi)
+
+        value.shrink()
+
+        cvalue = HCentered([value])
+        csymbol = HCentered([symbol])
+        width = max(cvalue.width, csymbol.width)
+        cvalue.hpack(width, 'exactly')
+        csymbol.hpack(width, 'exactly')
+
+        if shift:
+            vlist = Vlist([csymbol,                     # symbol
+                           Vbox(0, thickness * 2.),     # space
+                           cvalue                       # value
+                           ])
+
+            # Shift so the symbol sits in the same horizontal position
+            metrics = state.font_output.get_metrics(
+                state.font, mpl.rcParams['mathtext.default'],
+                r'\rightarrow', state.fontsize, state.dpi)
+
+            shift = ((csymbol.height) / 2. +
+                     (metrics.ymax + metrics.ymin) / 2. + thickness * 2.)
```

Comment:
```suggestion
            shift = (csymbol.height / 2 +
                     (metrics.ymax + metrics.ymin) / 2 + thickness * 2)
```

Reply:
I'm not sure what's wrong here, but it appears that `underset` is a little bit lower than `overset`:
```python
import matplotlib.pyplot as plt
plt.text(0.5, 0.5, r'Test of some $math \overset{foo}{\rightarrow} \underset{bar}{\rightarrow}$')
plt.savefig('foo.png', dpi=600)
```
![image](https://user-images.githubusercontent.com/302469/100972962-aee1aa00-3507-11eb-9ac2-70cfef52ff25.png)

Reply:
You're right, there were some logistic issues with the old approach:
![Screenshot from 2020-12-09 16-46-53](https://user-images.githubusercontent.com/43996118/101623269-349cb280-3a3e-11eb-90b0-de7d5e376cf8.png)
I made some changes, this is how it looks now:
![Screenshot from 2020-12-09 16-40-52](https://user-images.githubusercontent.com/43996118/101622786-81cc5480-3a3d-11eb-91dd-01c31ae961aa.png)
Alternatively, it'd be helpful to look at some characters which are not italicised or have all characters of same height:
![Screenshot from 2020-12-09 16-43-24](https://user-images.githubusercontent.com/43996118/101622938-b3ddb680-3a3d-11eb-8577-dad4185cd65c.png)
![Screenshot from 2020-12-09 16-50-30](https://user-images.githubusercontent.com/43996118/101623610-b5f44500-3a3e-11eb-9933-cc9a32910039.png)